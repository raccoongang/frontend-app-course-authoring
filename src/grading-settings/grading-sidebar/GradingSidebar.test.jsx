import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';

import GradingSidebar from '.';

const mockPathname = '/foo-bar';
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: mockPathname,
  }),
}));

describe('<GradingSidebar />', () => {
  const config = { STUDIO_BASE_URL: 'https://example.com' };
  const courseId = 'course123';
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <AppContext.Provider value={{ config }}>
          <IntlProvider locale="en">
            <GradingSidebar
              intl={injectIntl}
              courseId={courseId}
              proctoredExamSettingsUrl="link://to"
            />
          </IntlProvider>
        </AppContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
