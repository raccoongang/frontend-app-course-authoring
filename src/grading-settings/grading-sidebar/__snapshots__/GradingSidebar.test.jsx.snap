// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradingSidebar /> should match the snapshot 1`] = `
<aside
  className="help-sidebar"
>
  <div
    className="help-sidebar-about"
  >
    <h4
      className="help-sidebar-about-title"
    >
      What can I do on this page?
    </h4>
    <p
      className="help-sidebar-about-descriptions"
    >
      You can use the slider under Overall Grade Range to specify whether your course is pass/fail or graded by letter, and to establish the thresholds for each grade.
    </p>
    <p
      className="help-sidebar-about-descriptions"
    >
      You can specify whether your course offers students a grace period for late assignments.
    </p>
    <p
      className="help-sidebar-about-descriptions"
    >
      You can also create assignment types, such as homework, labs, quizzes, and exams, and specify how much of a student's grade each assignment type is worth.
    </p>
  </div>
  <hr />
  <div
    className="help-sidebar-other"
  >
    <h4
      className="help-sidebar-other-title"
    >
      Other Course Settings
    </h4>
    <nav
      aria-label="Other Course Settings"
      className="help-sidebar-other-links"
    >
      <ul
        className="p-0 mb-0"
      >
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/settings/details"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Schedule & Details
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/settings/grading"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Grading
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/course_team"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Course Team
          </a>
        </li>
        
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="https://example.com/group_configurations/course123"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Group Configurations
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/settings/advanced"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Advanced Settings
          </a>
        </li>
      </ul>
    </nav>
  </div>
</aside>
`;
